sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DistributionAndOrderingUtils.scala

```diff
@@ -32,23 +32,34 @@ object DistributionAndOrderingUtils {
     case write: RequiresDistributionAndOrdering =>
       val resolver = conf.resolver
 
-      val distribution = write.requiredDistribution match {
+      val numParts = write.requiredNumPartitions()
+      val (distribution, numPartitions) = write.requiredDistribution match {
```

Comment:
I don't think it is necessary to return `numParts` from the `match` statement here. It is no longer carried by the distribution.

Reply:
Missed one. Nice catch!